core/src/main/java/org/elasticsearch/action/admin/cluster/allocation/TransportClusterAllocationExplainAction.java

```diff
@@ -161,6 +166,9 @@ public static NodeExplanation calculateNodeExplanation(ShardRouting shard,
         if (node.getId().equals(assignedNodeId)) {
             finalDecision = ClusterAllocationExplanation.FinalDecision.ALREADY_ASSIGNED;
             finalExplanation = "the shard is already assigned to this node";
+        } else if (shard.unassigned() && hasPendingAsyncFetch) {
+            finalExplanation = "the shard state fetch is pending";
```

Comment:
I'll try to change the message to be clearer to the user (I agree it doesn't make as much sense right now)

I agree on the throttling one, I'll open a follow-up for this